com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateSubscribersStep.java

```diff
@@ -180,13 +179,7 @@ private CloudOrganization createDummyOrg(String orgName) {
     }
 
     private List<CloudApplication> removeDuplicates(List<CloudApplication> applications) {
-        Map<UUID, CloudApplication> applicationsMap = new LinkedHashMap<>();
-        for (CloudApplication application : applications) {
-            applicationsMap.put(application.getMetadata()
-                                           .getGuid(),
-                                application);
-        }
-        return new ArrayList<>(applicationsMap.values());
+        return new ArrayList<>(new LinkedHashSet<>(applications));
```

Comment:
This won't work - the above code assume we have different CloudApplication instances but with the same guid in the metadata. Fix this and add unit test if this wasn't caught.

Reply:
Can more that one application (with same guid) ever exist in the same space? 
Wasn't the point of the guid of the app that it was unique?
Are more that one instances of the same app treated as separate `CloudApplication` objects?

Reply:
discussed on other channel - please verify and revert